src/main/java/frc/robot/subsystems/ShooterSubsystem.java

```diff
@@ -0,0 +1,152 @@
+package frc.robot.subsystems;
+
+import com.revrobotics.CANSparkBase;
+import com.revrobotics.CANSparkLowLevel.MotorType;
+import com.revrobotics.CANSparkMax;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+
+public class ShooterSubsystem extends SubsystemBase {
+
+  private final CANSparkMax shooterTopMotor;
+  private final CANSparkMax shooterBottomMotor;
+
+  public ShooterSubsystem() {
+    shooterTopMotor =
+        new CANSparkMax(Constants.ShooterConstants.SHOOTER_MOTOR_TOP_ID, MotorType.kBrushless);
+    shooterBottomMotor =
+        new CANSparkMax(Constants.ShooterConstants.SHOOTER_MOTOR_BOTTOM_ID, MotorType.kBrushless);
+
+    shooterTopMotor.restoreFactoryDefaults();
+    shooterTopMotor.setSmartCurrentLimit(40);
+    shooterBottomMotor.restoreFactoryDefaults();
+    shooterBottomMotor.setSmartCurrentLimit(40);
+
+    shooterTopMotor.getPIDController().setP(Constants.ShooterConstants.SHOOTER_P);
+    shooterTopMotor.getPIDController().setI(Constants.ShooterConstants.SHOOTER_I);
+    shooterTopMotor.getPIDController().setD(Constants.ShooterConstants.SHOOTER_D);
+    shooterTopMotor.getPIDController().setFF(Constants.ShooterConstants.SHOOTER_FF);
+    shooterBottomMotor.getPIDController().setP(Constants.ShooterConstants.SHOOTER_P);
+    shooterBottomMotor.getPIDController().setI(Constants.ShooterConstants.SHOOTER_I);
+    shooterBottomMotor.getPIDController().setD(Constants.ShooterConstants.SHOOTER_D);
+    shooterBottomMotor.getPIDController().setFF(Constants.ShooterConstants.SHOOTER_FF);
+  }
+
+  // Fires the shooter.
+  public Command shooterFire(double distance) {
+    return startEnd(
+        () -> {
+          if (isShooterReady(distance)) {
+            setShooterSpeeds(calculateShooterSpeeds(distance));
+          }
+        },
+        () -> {});
+  }
+
+  // Idle shooter command; for default command purposes
+  public Command shooterIdle() {
+    return run(
+        () -> {
+          setShooterSpeeds(Constants.ShooterConstants.SHOOTER_IDLE_SPEEDS);
+        });
+  }
```

Comment:
Append .withName(...) to the returned Commands so we can track them in logs

Reply:
Resolved in 6372054